### DISCUSSION TITLE
[BUG]: Copy-PnpFile not working from Team Site Library to O365 Group Library - Returns nothing, but doesn't copy.

### DISCUSSION THREAD

**Discussion by jeremyhagan** [pnp/powershell#1289] [Q&A]
### Reporting an Issue or Missing Feature
When I use Copy-PnpFile to copy within a Doc library it works fine, but when trying to copy from the Root Team site to a Doc library on another site the file is not copied and no error is returned.

I've tried various combinations of including the leading "/" in the sourceUrl and specifying or omitting the -connection parameter in the command. 

Not using the -connection parameter or specifying the target site connection returns no error, but the file is not copied.

Omitting the leading "/" in the SourceUrl causes the error "The system cannot find the file specified"

Specifying the source site connection causes a return value with the message: The system cannot find the path specified. (Exception from HRESULT: 0x80070003)

### Expected behavior 
Expect the file to copy

### Actual behavior 
Nothing

### Steps to reproduce behavior
```
$AConnection = Connect-PnPOnline -Url "https://example.sharepoint.com/sites/a" -Credentials $Credential -ReturnConnection
$AContext = Get-PnPContext

$BConnection = Connect-PnPOnline -Url "https://example.sharepoint.com/" -Credentials $Credential -ReturnConnection
$BContext = Get-PnPContext

Copy-PnPFile -SourceUrl "/Resources/XXX/foo.docx" -TargetUrl "/sites/a/YYY/bar/targetfolder" -Force
```

### What is the version of the Cmdlet module you are running?
Copy-PnpFile version 1.8.0

### Which operating system/environment are you running PnP PowerShell on?
- [ X] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by veronicageek** [comment]
@jeremyhagan 
>trying to copy from the Root Team site to a Doc library on another site

Are you just trying to copy a document from a document library in `Site A` to a document library in `Site B`?

**Comment by ToddKlindt** [comment]
I was able to do this with 1.8.0. Here are the steps I used:
```
Connect-PnPOnline -Url https://m365x995492.sharepoint.com/
# Copy the file to another doclib in the root, just as a test
Copy-PnPFile -SourceUrl "/Shared Documents/Does this work.docx" -TargetUrl "/TestLib"
# Copy to a different site
Copy-PnPFile -SourceUrl "/Shared Documents/Does this work.docx" -TargetUrl "/sites/ContosoWorks/Shared Documents"

```
Here's how it looks:
![image](https://user-images.githubusercontent.com/2659747/138130711-6ba0f8ad-baee-4f34-90c0-43150f2c3ac5.png)

**Comment by jeremyhagan** [comment]
> I was able to do this with 1.8.0. Here are the steps I used:
> 
> ```
> Connect-PnPOnline -Url https://m365x995492.sharepoint.com/
> # Copy the file to another doclib in the root, just as a test
> Copy-PnPFile -SourceUrl "/Shared Documents/Does this work.docx" -TargetUrl "/TestLib"
> # Copy to a different site
> Copy-PnPFile -SourceUrl "/Shared Documents/Does this work.docx" -TargetUrl "/sites/ContosoWorks/Shared Documents"
> ```
> 
> Here's how it looks: ![image](https://user-images.githubusercontent.com/2659747/138130711-6ba0f8ad-baee-4f34-90c0-43150f2c3ac5.png)

Hi,
If I try like that (without the second connect statement to the target site) I get Access Denied. I can use the same account to copy documents within the library in sites/a

**Comment by jeremyhagan** [comment]
> @jeremyhagan
> 
> > trying to copy from the Root Team site to a Doc library on another site
> 
> Are you just trying to copy a document from a document library in `Site A` to a document library in `Site B`?

Yes, except that the source site is the default SP Team Site. I will try some more different combinations to see if I can isolate this further.

**Comment by jeremyhagan** [comment]
> I was able to do this with 1.8.0. Here are the steps I used:
> 
> ```
> Connect-PnPOnline -Url https://m365x995492.sharepoint.com/
> # Copy the file to another doclib in the root, just as a test
> Copy-PnPFile -SourceUrl "/Shared Documents/Does this work.docx" -TargetUrl "/TestLib"
> # Copy to a different site
> Copy-PnPFile -SourceUrl "/Shared Documents/Does this work.docx" -TargetUrl "/sites/ContosoWorks/Shared Documents"
> ```
> 
> Here's how it looks: ![image](https://user-images.githubusercontent.com/2659747/138130711-6ba0f8ad-baee-4f34-90c0-43150f2c3ac5.png)

@ToddKlindt , this is what happens to me:
```
Connect-PnPOnline -url http://mycompany.sharepoint.com/
Copy-PnPFile -SourceUrl "/Shared Documents/Jeremy Test.docx" -TargetUrl "/Resources"

Shared Documents/Jeremy Test.docx Microsoft.SharePoint.SPException: This operation is not supported. ---> System.Runtime.InteropServices.COMException: This operation is not supported

```
However, when I do the following it works:
```
Connect-PnPOnline -url http://mycompany.sharepoint.com/sites/myo365group/
Copy-PnPFile -SourceUrl "/sites/myo365group/MyLibrary/SubfolderA/Test.docx"  -TargetUrl "/sites/myo365group/MyLibrary/SubfolderB"
```

**Comment by ToddKlindt** [comment]
Is that the actual PowerShell you're using? I notice the URLs are https instead of HTTPS. 

How are you connecting? Regular username and password?

**Comment by jeremyhagan** [comment]
> Is that the actual PowerShell you're using? I notice the URLs are https instead of HTTPS.
> 
> How are you connecting? Regular username and password?

Typo (not copying and pasting). http instead of https, but otherwise yes. And normal username and password. I have almost got to the bottom of all but one issue, which I may have to leave. Will post a write up here when I know more.

**Comment by veronicageek** [comment]
Transferring to Discussions.

@jeremyhagan - As @ToddKlindt showed, there's no issue with the commands. And I also have no problem on my side copying from one site to the other.

> If I try like that (without the second connect statement to the target site) I get Access Denied

You don't need to have 2 different connections when copying from one site to the other, it'll switch automatically. Do you have access to the target site? 

Also, can you do the copy via the SPO UI?

**Comment by prossum** [comment]
If you get Access Denied on the target site, you do not have access with the account you use.
If you try to do the same manualy from the SharePoint UI, you start in the source-site and try to copy the file to the target site, you probably cannot select the target-site. And that's because you do not have access.
If you have a second account that has access to the target site, you need to download, reconnect and upload.
